invesalius/navigation/mtms.py

```diff
@@ -0,0 +1,115 @@
+import win32com.client
+
+import numpy as np
+import pandas as pd
+import random
+import time
+
+import invesalius.data.coregistration as dcr
+
+class mTMS():
+    def __init__(self):
+        mtms_path = 'C:\\mTMS\\Labview\\Builds\\mTMS 3.1 hack'
+        vipath = mtms_path + '\\mTMS ActiveX Server\\mTMS ActiveX Server.exe\\mTMS ActiveX Server.vi'
+        # Connect to the ActiveX server
+        mtms_app = win32com.client.Dispatch('MTMSActiveXServer.Application')
+        self.vi = mtms_app.getvireference(vipath)
+        # Log name
+        self.log_name = 'mtms_subject_00_run_0'
+        #self.vi.SetControlValue(self.log_name, 'Experiment 1a')
+        self.intensity = self.vi.GetControlValue('Get Intensity')
+        #self.intensity = 20
+
+        self.df = pd.DataFrame([], columns=["mTMS_target", "brain_target(nav)", "coil_pose(nav)", "intensity"])
+
+    def CheckTargets(self, coil_pose, brain_target_list):
+        for brain_target in brain_target_list:
+            distance = dcr.ComputeRelativeDistanceToTarget(target_coord=brain_target, img_coord=coil_pose)
+            offset = self.GetOffset(distance)
+            mTMS_target, mTMS_index_target = self.FindmTMSParameters(offset)
+            if not len(mTMS_index_target[0]):
+                print("Not possible to stimulate the target: ", offset)
+                return False
+        return True
+
+    def UpdateTargetSequence(self, coil_pose, brain_target_list):
+        if brain_target_list:
+            #Do I really need to check this? Or I can apply only the possible stimuli?
+            if self.CheckTargets(coil_pose, brain_target_list):
+                number_of_stim = 3
+                randomized_brain_target_list = brain_target_list.copy()
+                random.shuffle(randomized_brain_target_list)
+                for brain_target in randomized_brain_target_list:
+                    for x in range(number_of_stim):
+                        self.UpdateTarget(coil_pose, brain_target)
+                        time.sleep(random.randrange(300, 500, 1)/100)
+                self.SaveSequence()
+
+    def UpdateTarget(self, coil_pose, brain_target):
+        coil_pose_flip = coil_pose.copy()
+        brain_target_flip = brain_target.copy()
+        coil_pose_flip[1] = -coil_pose_flip[1]
+        brain_target_flip[1] = -brain_target_flip[1]
+        distance = dcr.ComputeRelativeDistanceToTarget(target_coord=coil_pose_flip, img_coord=brain_target_flip)
+        offset = self.GetOffset(distance)
+        print(offset)
```

Comment:
Are all of these prints needed? Should we have "Warning" dialogs?

Reply:
removed @6cbd68f0